src/storage/recovery/recovery_manager.cpp

```diff
@@ -144,15 +122,40 @@ void RecoveryManager::DeferRecordDeletes(terrier::transaction::timestamp_t txn_i
 
 uint32_t RecoveryManager::ProcessDeferredTransactions(terrier::transaction::timestamp_t upper_bound_ts) {
   auto txns_processed = 0;
+
   // If the upper bound is INVALID_TXN_TIMESTAMP, then we should process all deferred txns. We can accomplish this by
   // setting the upper bound to INT_MAX
   upper_bound_ts =
       (upper_bound_ts == transaction::INVALID_TXN_TIMESTAMP) ? transaction::timestamp_t(INT64_MAX) : upper_bound_ts;
   auto upper_bound_it = deferred_txns_.upper_bound(upper_bound_ts);
 
+  // For recording recovery throughput metric
+  uint64_t elapsed_us = 0;
+  uint64_t total_elapsed_us = 0;
+  uint64_t num_txns = 0;
+  uint64_t num_bytes = 0;
   for (auto it = deferred_txns_.begin(); it != upper_bound_it; it++) {
-    ProcessCommittedTransaction(*it);
+    {
+      common::ScopedTimer<std::chrono::microseconds> scoped_timer(&elapsed_us);
+      num_bytes += ProcessCommittedTransaction(*it);
+    }
     txns_processed++;
+
+    // Update metrics
+    total_elapsed_us += elapsed_us;
+    num_txns += 1;
+
+    // Check if we need to export throughput metric
+    if (std::chrono::milliseconds(total_elapsed_us) >= recovery_metric_interval_ || std::next(it) == upper_bound_it) {
```

Comment:
@mbutrovich I want your input on this. Should components export metrics as soon as they have data collected, or should they input them on an interview basis (i.e. output a metrics every second for the processing done for the last second)

Reply:
Lets take out the interval stuff and just output a metric with the same logic below